daskms/writes.py

```diff
@@ -644,13 +644,24 @@ def _write_datasets(
 
             inlinable_arrays = [row_order]
 
-            if (
-                row_order.shape[0] != array.shape[0]
-                or row_order.chunks[0] != array.chunks[0]
-            ):
-                raise ValueError(
-                    f"ROWID shape and/or chunking does " f"not match that of {column}"
-                )
+            if np.isnan(row_order.shape[0]):  # Dealing with nan chunks/dims.
+                if not (
+                    np.isnan(row_order.shape[0]) == np.isnan(array.shape[0])
+                    and len(row_order.chunks[0]) == len(array.chunks[0])
+                ):
+                    raise ValueError(
+                        f"ROWID shape and/or chunking does "
+                        f"not match that of {column}"
+                    )
+            else:
+                if (
+                    row_order.shape[0] != array.shape[0]
+                    or row_order.chunks[0] != array.chunks[0]
+                ):
+                    raise ValueError(
+                        f"ROWID shape and/or chunking does "
+                        f"not match that of {column}"
+                    )
```

Comment:
How about?

```python
if (np.isnan(row_order.shape[0]) != np.isnan(array.shape[0]) or
   row_order.shape[0] != array.shape[0] or
   any(np.isnan(a) != np.isnan(b)  or a != b for a, b in zip(row_order.chunks[0],array.chunks[0]))):

      raise ValueError(...)
```

Reply:
Hmmm, more concise but quite confusing. Will think about it for a moment.

Reply:
I like this compact expression but getting it to do the correct thing in call cases is difficult.

Reply:
> This doesn't work as written, as the case second conditional will always evaluate to True for nan chunks.

Good catch. How about this?

```python
if not (np.isnan(row_order.shape[0]) if np.isnan(array.shape[0]) else row_order.shape[0] == array.shape[0] and
   all(np.isnan(a) if np.isnan(b)  else a == b for a, b in zip(row_order.chunks[0],array.chunks[0]))):

      raise ValueError(...)

```

Reply:
I think it works but I have no idea how to format it. Hahaha.

Reply:
Ok, applied this change. If you are happy, go ahead and merge.

Reply:
Yeah apologies, I'm now using a pre-commit hook everywhere.  For future reference

```
(venv) $ pip install pre-commit
(venv) $ pre-commit install
(venv) $ pre-commit run --all-files
```

I should probably put this in the docs somewhere